src/main/java/co/cask/hydrator/plugin/RecordSplitter.java

```diff
@@ -142,10 +142,12 @@ public void validate(Schema inputSchema) throws IllegalArgumentException {
           ? inputFieldSchema.getNonNullable().getType()
           : inputFieldSchema.getType();
         if (!(inputFieldType == Schema.Type.STRING)) {
-          throw new IllegalArgumentException("Source field: " + fieldToSplit +
-                                             " must be of type string. It is type: " +
-                                             inputFieldType.name());
+          throw new IllegalArgumentException(String.format("Source field: %s must be of type string. It is type: %s",
+                                                           fieldToSplit, inputFieldType.name()));
         }
+      } else {
+        throw new IllegalArgumentException(String.format("Source field: %s must be of type string. It is type: %s",
+                                                         fieldToSplit, inputFieldSchema.getType().name()));
```

Comment:
inputFieldSchema.getType() will always be a union if it is complex nullable.
If we are only supporting Strings you dont have to check if it is a simpleOrNullableSimple(). as string will always be simple type. We can do

Schema type = inputFieldSchema.isNullable() : inputFieldSchema.getNonNullable().getType ?  inputFieldSchema.getType;
if ( type != Schema.Type.STRING) {
    throw exception 
 }